haralyzer/sub_classes.py

```diff
@@ -0,0 +1,161 @@
+"""Creates the Request and Response sub class that are used by each entry"""
+from cached_property import cached_property
+
+
+class Request(object):
+    """Request object for an HarEntry"""
+    def __init__(self, entry):
+        self.raw_entry = entry
+
+    def get_header_value(self, name):
+        """
+        Returns the header value of the header defined in ``name``
+
+        :param name: ``str`` name of the header to get the value of
+        """
+        for x in self.raw_entry["headers"]:
+            if x["name"].lower() == name.lower():
+                return x["value"]
+
+    # Root Level values
+
+    @cached_property
+    def bodySize(self):
+        return self.raw_entry["bodySize"]
+
+    @cached_property
+    def method(self):
+        return self.raw_entry["method"]
+
+    @cached_property
+    def url(self):
+        return self.raw_entry["url"]
+
+    @cached_property
+    def headers(self):
+        return self.raw_entry["headers"]
+
+    @cached_property
+    def httpVersion(self):
+        return self.raw_entry["httpVersion"]
+
+    @cached_property
+    def queryString(self):
+        return self.raw_entry["queryString"]
+
+    @cached_property
+    def headersSize(self):
+        return self.raw_entry["headersSize"]
+
+    @cached_property
+    def cookies(self):
+        return self.raw_entry["cookies"]
+
+    # Header Values
+
+    @cached_property
+    def host(self):
+        return self.get_header_value("Host")
+
+    @cached_property
+    def userAgent(self):
+        return self.get_header_value("User-Agent")
+
+    @cached_property
+    def accept(self):
+        return self.get_header_value("Accept")
+
+    @cached_property
+    def language(self):
+        return self.get_header_value("Accept-Language")
+
+    @cached_property
+    def encoding(self):
+        return self.get_header_value("Accept-Encoding")
+
+    @cached_property
+    def cacheControl(self):
+        return self.get_header_value("Cache-Control")
+
+
+class Response(object):
+    """Response object for a HarEntry"""
+    def __init__(self, entry):
+        self.raw_entry = entry
+
+    def get_header_value(self, name):

```

Comment:
Rule of 3 here. We should find a way to provide this method only once and propagate it to all classes that need it (a mixin or base class).

Reply:
Did not know about Mixins. Will a Mixin for that function.